### ISSUE TITLE
RCS1031

### ISSUE THREAD

**Issue by jnyrup** [JosefPihrt/Roslynator#44]
I agree that this is a terrible misuse of local scopes to circumvent bad coding style...
RCS1031 is triggered on both pair of braces, but the code becomes invalid if any local scope is removed.
```
{
	int a = 1;
}
{
	int a = 1;
}
```

**Comment by JosefPihrt** [comment]
I tried to figure out how verify that removing the braces will not break the code, but it seems to be way too complex. So I think that only reliable solution is to delete this analyzer.

**Comment by jnyrup** [comment]
In order to remove sub scope `b` from scope `a`, all variables declared directly in `b` must not be declared in any other sub scopes of `a`.

````
{ // scope a
    { // scope b
        ...
    }
    { // scope c
        ...
    }
        ...
}
````

Could one do something like the following pseudo code?
````
bool CanRemoveScope(Scope s)
  foreach Declaration d in s:
    foreach Scope s' in s.ParentScope except s:
      if s' or subscope of s' contains d:
        return false

  return true
````

**Comment by JosefPihrt** [comment]
The example you provided is only one special case of many.

Generally speaking, it is not about two 'subscopes', but rather about scope (block) and parent scope (parent block).

To make things even worse there are cases where there will be no error reported if you remove the braces but the semantics will change. Look at the following example:

```csharp
    public class Foo
    {
        private object f;

        public void Bar()
        {
            {
                object f = null;
            }

            // If the braces are removed 'f' will change from local to field (but not error will be reported).
            f = null;

            // This one is fine, because it is qualified.
            this.f = null;
        }
    }
```

This example is only one of many cases that can occur.

**Comment by jnyrup** [comment]
I stand corrected, this does seem very complex.
